File: CommonDataModel/src/main/java/com/kmitl/b_mw/common/nodeModel/ComputeUnit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.kmitl.b_mw.common.nodeModel;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 *
 * @author B_MW (Wai & May)
 */
public class ComputeUnit extends BasicNode {

    final String nodeType = "ComputeUnit";

    String parent = null;
    String state = null;
    List<String> devices = new ArrayList<>();

    public String getParent() {
        return parent;
    }

    public void setParent(String parent) {
        this.parent = parent;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public List<String> getDevices() {
        return devices;
    }

    public void setDevices(List<String> devices) {
        this.devices = devices;
    }

}

File: Reader/lib/model/common/ComputeUnit.js
var BasicNode = require("./BasicNode");

module.exports = (function () {
  function ComputeUnit (id, key, name, validFrom, validTo) {
    BasicNode.call(this, id, key, name ,"ComputeUnit", validFrom, validTo);
    this.parent = null;
    this.state = null;
    this.devices = [];
  }

  ComputeUnit.prototype = BasicNode.prototype;
  ComputeUnit.prototype = BasicNode.prototype;
  ComputeUnit.prototype.getParent = getParent;
  ComputeUnit.prototype.setParent = setParent;
  ComputeUnit.prototype.getState = getState;
  ComputeUnit.prototype.setState = setState;
  ComputeUnit.prototype.getDevices = getDevices;
  ComputeUnit.prototype.setDevices = setDevices;

  function getParent() {
      return this.parent;
  }

  function setParent (parent) {
      this.parent = parent;
  }

  function getState () {
      return this.state;
  }

  function setState (state) {
      this.state = state;
  }

  function getDevices () {
      return this.devices;
  }

  function setDevices (devices) {
      this.devices = devices;
  }

  return ComputeUnit;
})();
